server/option.go

```diff
@@ -127,15 +127,22 @@ func (o *scheduleOption) GetSchedulers() SchedulerConfigs {
 func (o *scheduleOption) AddSchedulerCfg(tp string, args []string) error {
 	c := o.load()
 	v := c.clone()
-	for _, schedulerCfg := range v.Schedulers {
+	for i, schedulerCfg := range v.Schedulers {
 		// comparing args is to cover the case that there are schedulers in same type but not with same name
 		// such as two schedulers of type "evict-leader",
 		// one name is "evict-leader-scheduler-1" and the other is "evict-leader-scheduler-2"
-		if reflect.DeepEqual(schedulerCfg, SchedulerConfig{tp, args}) {
+		if reflect.DeepEqual(schedulerCfg, SchedulerConfig{tp, args, false}) {
+			return nil
+		}
+
+		if reflect.DeepEqual(schedulerCfg, SchedulerConfig{tp, args, true}) {
```

Comment:
can you add fields in the struct like L145 does?

Reply:
why true here and then set false later?

Reply:
Because add a schedule means re-enabling this schedule